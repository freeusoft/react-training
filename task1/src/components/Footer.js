import React, { PureComponent } from 'react';
import ReactDOM from 'react-dom';

export default class Footer extends PureComponent {
  render() {
    return (
      <div className="footer">
        Footer component (React.PureComponent)
      </div>
    )
  }
}
