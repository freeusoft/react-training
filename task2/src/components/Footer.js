import React, { PureComponent } from 'react';

export default class Footer extends PureComponent {
  render() {
    return (
      <div className="footer">
        Footer component (React.PureComponent)
      </div>
    )
  }
}
