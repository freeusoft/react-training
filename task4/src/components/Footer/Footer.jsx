import React, { Component } from 'react'

export default class Footer extends Component {
  render () {
    return (
      <div className='footer'>
        <span className='red-font'>MovieDB</span>
        <span className='env-mode'>{process.env.NODE_ENV === 'production' ? 'prod' : 'dev'}</span>
      </div>
    )
  }
}
