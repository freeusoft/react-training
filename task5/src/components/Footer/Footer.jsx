import React, { Component } from 'react'

export default class Footer extends Component {
  render () {
    return (
      <div className='footer'>
        <span className='title red-font'>MovieDB</span>
        <span className='env-mode'>{this.props.node_env === 'production' ? 'prod' : 'dev'}</span>
      </div>
    )
  }
}
