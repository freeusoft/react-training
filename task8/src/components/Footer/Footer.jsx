// @flow
import React, { Component } from 'react'
import classNames from 'classnames'
import injectSheet from 'react-jss'

const styles = {
  footer: {
    height: '60px',
    background: {
      color: '#424242'
    },
    padding: {
      left: '50px',
      top: '17px'
    },
    font: {
      weight: 'bold',
      size: '14pt'
    }
  },
  redFont: {
    color: '#f55363'
  },
  envMode: {
    float: 'right',
    padding: {
      right: '50px'
    },
    color: 'gray'
  }
}

class Footer extends Component<Object, Object> {
  render () {
    const { classes } = this.props
    return (
      <div className={classes.footer}>
        <span className={classNames(classes.title, classes.redFont)}>MovieDB</span>
        <span className={classes.envMode}>{this.props.node_env === 'production' ? 'prod' : 'dev'}</span>
      </div>
    )
  }
}

export default injectSheet(styles)(Footer)