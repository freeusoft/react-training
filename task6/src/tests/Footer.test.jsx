import React from 'react'
import Footer from '../components/Footer/Footer'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

describe('Footer component test suite', ()=>{
  test('Footer snapshot renders correctly', () => {
    const tree = renderer.create(
      <Footer />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Footer caption should have text equal', () => {
    const wrapper = shallow(
      <Footer />
    )
    const divTitle = wrapper.find('.title')
    expect(divTitle.text()).toBe('MovieDB')
  })

  test('Footer env mode must be "dev"', () => {
    const wrapper = shallow(
      <Footer />
    );
    const divEnv = wrapper.find('.env-mode')
    expect(divEnv.text()).toBe('dev')
  })

  test('Footer env mode must be "prod"', () => {
    const wrapper = shallow(
      <Footer node_env='production'/>
    )
    const divEnv = wrapper.find('.env-mode')
    expect(divEnv.text()).toBe('prod')
  })

}) 
