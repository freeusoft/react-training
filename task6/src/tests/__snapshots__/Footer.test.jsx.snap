// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component test suite Footer snapshot renders correctly 1`] = `
<div
  className="footer"
>
  <span
    className="title red-font"
  >
    MovieDB
  </span>
  <span
    className="env-mode"
  >
    dev
  </span>
</div>
`;
